### PR TITLE
DM-43357: Add support for using catalog PSF parameters

### PR DESCRIPTION
**Pull request by taranu**
In practice this is for DoubleShapelet PSF fits specifically, but generally works for reading any kind of PSF parameters from a catalog.

### CODE REVIEW

**Review by cmsaunders — APPROVED** [review]
I have a few minor comments. It's hard to follow what's being done in total on this PR, but the individual changes look reasonable. There are a lot of lingering TODOs, but I guess since this is part of `lsst-dm`, you aren't obligated to resolve them.

File: pipelines/coaddQualityCore.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+description: |
+  Tier1 (core) atools and metrics to assess MultiProFit fit quality
+parameters:
+  include_expdev: False
+  include_ser: False
+  include_ser_fixedcen: False
+  size_mag_plot_xLims: [17, 29]
+  size_mag_plot_yLims: [-4, 3]
+tasks:
+  analyzeObjectTableCore:
```
[the remark this answers is not in the document]
Comment (taranu) [reply pipelines/coaddQualityCore.yaml:10]: Hmm, I left it like this to be run separately for now and to be eventually merged into the atools pipeline of the same name without needing to register new dataset types, though the ship has kind of sailed on avoiding flooding the registry with thousands of atools dataset types.

File: python/lsst/meas/extensions/multiprofit/fit_coadd_psf.py [hunk 2]
```diff
@@ -81,6 +87,53 @@ def check_source(self, source, config):
...
+        if limits_x is None:
+            limits_x = g2f.LimitsD(0, n_cols)
+        if limits_y is None:
+            limits_y = g2f.LimitsD(0, n_rows)
+
+        for component, config_comp in zip(
+            config_data.components.values(), config_data.component_configs.values()
+        ):
+            centroid = component.centroid
+            if centroid not in centroids:
```
[the remark this answers is not in the document]
Comment (taranu) [reply python/lsst/meas/extensions/multiprofit/fit_coadd_psf.py:122]: Right now yes, there's only one centroid object shared between all components in the model, but in principle they could all be different. This just avoids setting the same centroid repeatedly, which is harmless but also pointless.
